merlin/dag/executors.py

```diff
@@ -156,12 +160,17 @@ def _transform_data(self, node, input_data, capture_dtypes=False):
             # Update or validate output_data dtypes
             for col_name, output_col_schema in node.output_schema.column_schemas.items():
                 col_series = output_data[col_name]
-                col_dtype = col_series.dtype
+                col_dtype = col_series.dtype  # tf.int32
                 is_list = is_list_dtype(col_series)
 
                 if is_list:
                     col_dtype = list_val_dtype(col_series)
+                if hasattr(col_dtype, "as_numpy_dtype"):
```

Comment:
This is where having a "merlin" dtype would have been very helpful. We can avoid this code kludge.